src/rgw/driver/d4n/d4n_policy.cc

```diff
@@ -475,7 +475,7 @@ void LFUDAPolicy::update(const DoutPrefixProvider* dpp, std::string& key, uint64
   weightSum += ((localWeight < 0) ? 0 : localWeight);
 }
 
-void LFUDAPolicy::update_dirty_object(const DoutPrefixProvider* dpp, std::string& key, std::string version, bool deleteMarker, uint64_t size, time_t creationTime, const rgw_user user, std::string& etag, const std::string& bucket_name, const std::string& bucket_id, const rgw_obj_key& obj_key, uint32_t refcount_flag, optional_yield y, std::string& restore_val)
+void LFUDAPolicy::update_dirty_object(const DoutPrefixProvider* dpp, std::string& key, std::string version, bool deleteMarker, uint64_t size, double creationTime, const rgw_user user, std::string& etag, const std::string& bucket_name, const std::string& bucket_id, const rgw_obj_key& obj_key, uint32_t refcount_flag, optional_yield y, std::string& restore_val)
```

Comment:
why double?  shouldn't we be using a time type?  e.g., ceph::real_time or something?

Reply:
@mattbenjamin : we changed from time_t to double since it has more precision than time_t, which was causing problems while calculating a score for versions in ordered set in case. We can revisit this later to figure out a better, for now I will let it be double.